usaspending_api/common/tests/unit/test_fiscal_year_helpers.py

```diff
@@ -69,41 +72,48 @@ def test_dates_are_fiscal_year_bookends():
 
 
+@pytest.mark.django_db
 def test_calculate_last_completed_fiscal_quarter():
-    """
-    FY2000 Q1 == 1999-10-01 - 1999-12-31 available after 2000-02-14
-    FY2000 Q2 == 2000-01-01 - 2000-03-31 available after 2000-05-15
-    FY2000 Q3 == 2000-04-01 - 2000-06-30 available after 2000-08-14
-    FY2000 Q4 == 2000-07-01 - 2000-09-30 available after 2000-11-14
-    """
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(1999, 10, 1)) is None
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(1999, 11, 1)) is None
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(1999, 12, 1)) is None
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 1, 1)) is None
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 2, 1)) is None
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 2, 14)) is None
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 2, 15)) == 1
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 3, 1)) == 1
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 4, 1)) == 1
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 5, 1)) == 1
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 5, 15)) == 1
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 5, 16)) == 2
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 6, 1)) == 2
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 7, 1)) == 2
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 8, 1)) == 2
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 9, 1)) == 3
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 10, 1)) == 3
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 11, 1)) == 3
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2000, 12, 1)) == 4
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2001, 1, 1)) == 4
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2001, 2, 1)) == 4
-    assert fyh.calculate_last_completed_fiscal_quarter(2000, date(2001, 3, 1)) == 4
-
-    # Test a fiscal year WAY in the future.  We should definitely have flying cars by then.
-    assert fyh.calculate_last_completed_fiscal_quarter(2010, date(2001, 3, 1)) is None
-
-    # And one test using the default as_of_date just to make sure nothing catches on fire.
-    assert fyh.calculate_last_completed_fiscal_quarter(2000) == 4
+    now = datetime.now(timezone.utc)
+    yesterday = now + timedelta(days=-1)
+    tomorrow = now + timedelta(days=1)
+    current_fy = fyh.generate_fiscal_year(now)
+
+    mommy.make(
+        "submissions.DABSSubmissionWindowSchedule",
+        submission_fiscal_year=2000,
+        submission_reveal_date=now,
```

Comment:
Any reason the fiscal_year is not the same as the year of the reveal date?

Reply:
No, but they don't have to be either. As long as the requested year matches and reveal_date is "now or earlier", it produces expected results.